rustworkx-core/src/token_swapper.rs

```diff
@@ -185,7 +186,13 @@ where
             Ok(res) => Ok(res.len()),
             Err(e) => Err(*e),
         })
-        .unwrap()
+        .unwrap()?;
+        // Map NodeIndex swaps to NodeId swaps
+        let swaps = result
+            .into_iter()
+            .map(|(a, b)| (self.rev_node_map[&a], self.rev_node_map[&b]))
+            .collect();
+        Ok(swaps)
```

Comment:
While this does return the swaps in the `NodeId` form, it adds an extra heap allocation to do it, which should be avoidable - in the parentheses in the original issue, I'd given a couple of suggestions for how (I think - I didn't do it completely) it should be possible to modify the internal tracking structures so that the natural output is `G::NodeId`, rather than needing an explicit remapping.